### ISSUE TITLE
[cloud] C6: rules-only classifier for cloud — lazy-import torch/SetFit

### ISSUE THREAD

**Issue by yadava5** [yadava5/jobtracker#17]
## Problem

`backend/jobtracker/classifier/hybrid.py` eagerly imports sentence-transformers, torch, and setfit at module top. When the cloud FastAPI app eventually mounts the classifier router (future issue), Vercel's function build will attempt to install torch (~800 MB) and blow the 50 MB zipped / 250 MB unzipped Hobby function-size limit. Even on Pro, the classifier isn't useful in a serverless function with a 60s wall clock — cold-starting sentence-transformers alone can exceed it.

## Why it matters

Without this fix, C6 blocks all cloud classification endpoints, which in turn blocks C7 cron, C11 applications list (needs classified emails), and C16 review queue. Rules-only coverage is sufficient for v1 because rules handle the majority of well-formed lifecycle emails; user-corrected labels still persist to `TrainingData` and sync back to macOS where full SetFit remains canonical.

## Real cause / current gap

- `backend/jobtracker/classifier/hybrid.py` has `from jobtracker.classifier.embeddings import get_embedding_classifier` and `from jobtracker.classifier.setfit_model import get_setfit_classifier` at module top.
- Both imports transitively pull in torch/sentence-transformers/setfit.
- `backend/jobtracker/classifier/__init__.py` exports `get_classifier()` which returns the hybrid classifier — so any cloud import of the classifier module triggers the heavy chain.
- There is no config-driven branch to opt out.

## Evidence

Plan agent research: classifier/hybrid.py:15-30 contains eager imports. Repo-wide grep of `get_classifier` finds call sites in `api/classification.py`, `api/emails.py`, `api/applications.py` — all transitively heavy when imported.

## Acceptance criteria

- [ ] `classifier/hybrid.py` lazy-imports `embeddings` and `setfit_model` — imports occur INSIDE method bodies, not at module top.
- [ ] When `settings.deployment == "cloud"`, `hybrid.classify()` short-circuits after the rules layer (does NOT load embeddings or SetFit).
- [ ] `settings.deployment == "cloud"` implicitly sets `settings.lite_mode = True` (or equivalent effect).
- [ ] A subprocess-isolated test verifies `from jobtracker.classifier import get_classifier; get_classifier()` under `JOBTRACKER_DEPLOYMENT=cloud` does NOT import torch/sentence-transformers/setfit.
- [ ] Existing desktop classifier tests unchanged (rules + embeddings + setfit paths all pass).
- [ ] Cloud classifier returns `{category, confidence, method: "rules"}` for rules hits and `{category: "other", confidence: 0.0, method: "rules"}` for rules misses (no further layers).
- [ ] Docs: `docs/WEB_ARCHITECTURE.md` classifier section updated.
- [ ] Existing backend-ci classifier gate (rules v3, hybrid v3 deterministic) continues to pass.

## Validation plan

```
# Desktop: all layers still work
.venv311/bin/pytest tests/test_evaluate_classifier.py tests/test_setfit_model.py -q

# Cloud: rules-only shortcircuit
JOBTRACKER_DEPLOYMENT=cloud .venv311/bin/pytest tests/test_main_cloud.py -q
JOBTRACKER_DEPLOYMENT=cloud python -c "from jobtracker.classifier import get_classifier; c = get_classifier(); print(c)"

# Subprocess import-hygiene guard (extend test_main_cloud.py)
# Assert torch/sentence-transformers/setfit NOT in sys.modules after
# a cloud classifier call.

# Full existing backend suite
.venv311/bin/pytest tests -q
```

## Dependencies

Blocked by: #14 (C1 shim). Blocks: C7, C11, C13, C16.

## Out of scope (this issue)

- Remote SetFit inference via Modal / HF Inference (deferred; document in `docs/ML_STRATEGY.md`).
- Training loop on cloud (impossible on Vercel's 60s budget).
- Changes to the existing CI rules v3 / hybrid v3 gates (those stay desktop-path tests).

**Comment by yadava5** [comment]
Closed by #19 — merged to integration/web-migration.
